daemon/cmd/daemon.go

```diff
@@ -217,6 +218,9 @@ type Daemon struct {
 	// statedb for implementing /statedb/dump
 	db statedb.DB
 
+	// authCertificateProvider for reporting the status of the auth system certificates
+	authCertificateProvider certs.CertificateProvider
```

Comment:
IMO it would be better if status reporting doesn't depend on internals of the auth module. It's a current implementation detail that there is only one auth handler which makes use of a certificate provider - hence there is only one of them. But this doesn't need to be the case.

It would be nicer if the auth module itself would report the status and aggregate all the details of its internals. In this case the authmanager would expose the status and would fetch the status from its internals (e.g. calling the authhandlers; that itself would call the certificateprovider, ...)